tests/accelerators/test_hpu.py

```diff
@@ -0,0 +1,375 @@
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from tests.helpers.boring_model import BoringModel
+from tests.helpers.datamodules import ClassifDataModule
+from tests.helpers.runif import RunIf
+from tests.helpers.simple_models import ClassificationModel
+
+if _HPU_AVAILABLE:
+    import habana_frameworks.torch.core as htcore
+
+    os.environ["PL_TORCH_DISTRIBUTED_BACKEND"] = "hccl"
+
+
+class HPUModel(BoringModel):
+    def training_step(self, batch, batch_idx):
+        output = self(batch)
+        loss = self.loss(batch, output)
+        return loss
+
+    def validation_step(self, batch, batch_idx):
+        output = self(batch)
+        loss = self.loss(batch, output)
+        return loss
+
+    def test_step(self, batch, batch_idx):
+        output = self(batch)
+        loss = self.loss(batch, output)
+        return loss
+
+    def training_epoch_end(self, outputs) -> None:
+        pass
+
+    def validation_epoch_end(self, outputs) -> None:
+        pass
+
+    def test_epoch_end(self, outputs) -> None:
+        pass
+
+
+class HPUClassificationModel(ClassificationModel):
+    def training_step(self, batch, batch_idx):
+        x, y = batch
+        logits = self(x)
+        loss = F.cross_entropy(logits, y)
+        return loss
+
+    def validation_step(self, batch, batch_idx):
+        x, y = batch
+        logits = self(x)
+        acc = self.accuracy(logits, y)
+        return acc
+
+    def test_step(self, batch, batch_idx):
+        x, y = batch
+        logits = self(x)
+        acc = self.accuracy(logits, y)
+        return acc
+
+    def accuracy(self, logits, y):
+        acc = torch.sum(torch.eq(torch.argmax(logits, -1), y).to(torch.float32)) / len(y)
+        return acc
+
+    def validation_epoch_end(self, outputs) -> None:
+        self.log("val_acc", torch.stack(outputs).mean())
+
+    def test_epoch_end(self, outputs) -> None:
+        self.log("test_acc", torch.stack(outputs).mean())
+
+
+def test_availability():
+    assert HPUAccelerator.is_available()
+
+
+@pytest.mark.skipif(_HPU_AVAILABLE, reason="test requires non-HPU machine")
+def test_fail_if_no_hpus(tmpdir):
+    with pytest.raises(MisconfigurationException, match="HPU Accelerator requires HPU devices to run"):
+        Trainer(default_root_dir=tmpdir, accelerator="hpu", devices=1)
+
+    with pytest.raises(MisconfigurationException, match="HPU Accelerator requires HPU devices to run"):
+        Trainer(default_root_dir=tmpdir, devices=1, accelerator="hpu")
+
+
+@RunIf(hpu=True)
+def test_accelerator_selected(tmpdir):
+    assert HPUAccelerator.is_available()
+    trainer = Trainer(default_root_dir=tmpdir, accelerator="hpu", devices=1)
+    assert isinstance(trainer.accelerator, HPUAccelerator)
+
+
+@RunIf(hpu=True)
+def test_no_warning_plugin(tmpdir):
+    with pytest.warns(None) as record:
+        Trainer(default_root_dir=tmpdir, max_epochs=1, strategy=HPUStrategy(device=torch.device("hpu")))
+    assert len(record) == 0
```

Comment:
This will fail even when unrelated warnings are raised. Could we filter it with something like its message?

Reply:
This is just to check the behavior without explicit plugin. No warnings are expected. We can remove this if nor required as we just followed ipu tests